mount/mount.c

```diff
@@ -23,6 +30,42 @@ static char* get_fs_type() {
 	return fs_type;
 }
 
+static int fill_ipv4_sockaddr(const char *hostname, struct sockaddr_in *addr)
+{
+	struct hostent *hp;
+	addr->sin_family = AF_INET;
+
+	if (inet_aton(hostname, &addr->sin_addr))
+		return 0;
+	if ((hp = gethostbyname(hostname)) == NULL) {
+		int err_ret = h_errno;
+		printf("titus-mount: can't get address for %s: %s\n", hostname, hstrerror(err_ret));
+		return -1;
+	}
+	if (hp->h_length > (int)sizeof(struct in_addr)) {
+		printf("titus-mount: got bad hp->h_length");
+		hp->h_length = sizeof(struct in_addr);
+	}
+	memcpy(&addr->sin_addr, hp->h_addr, hp->h_length);
+	return 0;
+}
+
+static int get_my_ipv4addr(char *ip_addr, int len)
+{
+	char myname[1024];
+	struct sockaddr_in myaddr;
+
+	if (gethostname(myname, sizeof(myname))) {
```

Comment:
I think the original approach / implementation of using a UDP socket was better. It took routing into account.

Reply:
Doing the UDP socket to establish the source local ip?
Why do we really need that? Won't the kernel pick the right ip (take routing into account) when establishing the connection anyway?